models/note.rb

```diff
@@ -0,0 +1,17 @@
+class Note < ActiveRecord::Base
+
+  has_many :taggings
+  has_many :tags, through: :taggings
+
+  validates :title, presence: true
+  validates :body, presence: true
+
+  def as_json(arg = nil)
```

Comment:
Since you defined this, you likely could have simplified your controller actions considerably